### PR TITLE
gh-pages: Add a robots.txt to exclude old docs for SEO

### PR DESCRIPTION
**Pull request by mortenpi**
The old docs do not define canonical URLs and can show up in Google searches even though more up to date docs are available. Having a `robots.txt` will hopefully prevent the old docs from showing up in searches in the future.

Just excluding these from indexing is easier, and possible more reliable, than adding canonical URLs to the old docs. Having someone more familiar with search engine optimization give their opinion on this would be handy though.

Note: this PR is against `gh-pages`.

cc @fredrikekre

### CODE REVIEW

**Comment by leto81** [comment]
I propose ''search engines'' instead of ''Google et al.''

File: robots.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Make sure that Google et.al. don't index the old documentation which did not
```
Comment (leto81) [inline robots.txt]: Make sure that search engines don't index [...]

**Comment by KristofferC** [comment]
Merge?

**Comment by StefanKarpinski** [comment]
I'm not sure what the criteria for merging this kind of PR are but we badly need this so @mortenpi, please go for it if you feel it's ready.

**Comment by vchuravy** [comment]
I am not sure that this is the right approach since we don't want to disallow indexing of the old pages. e.g googling `julia docs 0.3.0` should work? Maybe I am missunderstanding `robots.txt`.

Maybe it makes more sense to add a "canonical meta tag" or a sitemap?

**Comment by mortenpi** [comment]
Don't canonical URLs have the exact same effect in the end actually? Adding the necessary `<meta>` tags to the old docs would mean modifying each HTML file. And sitemaps won't allow us to exclude old docs, right?

I am wondering if adding `release-0.x` URLs here would also make sense. While they are 404-ing now, they still show up on Google and maybe robots.txt would make them de-index quicker.

**Comment by mortenpi** [comment]
Bump. Note that I can't merge it myself.

**Pull request merged**